packages/contracts/contracts/CollateralManagerV2.sol

```diff
@@ -0,0 +1,506 @@
+
+    //use getBundleInfo instead
+
+    function getCollateralInfo(uint256 _bidId)
+        public
+        view
+        returns (Collateral[] memory infos_)
+    {
+        uint256 count = _committedBidCollateral[_bidId].count;
+        infos_ = new Collateral[](count);
+
+        for (uint256 i = 0; i < count; i++) {
+            infos_[i] = _committedBidCollateral[_bidId].collaterals[i];
+        }
+    }
+
+    /**
+     * @notice Gets the collateral asset amount for a given bid id on the TellerV2 contract.
+     * @param _bidId The ID of a bid on TellerV2.
+     * @param _collateralAddress An address used as collateral.
+     * @return amount_ The amount of collateral of type _collateralAddress.
+     */
+    function getCollateralAmount(uint256 _bidId, address _collateralAddress)
+        public
+        view
+        returns (uint256 amount_)
+    {
+
+        uint256 bundleId = _collateralBundleIdForBid[_bidId];
+
+        Collateral memory token_data = getTokenOfBundle(bundleId, 0); // first slot
+        
+        if (token_data._collateralAddress != _collateralAddress) return 0; // not as expected
+
+        amount_ = token_data._amount;
+    }
+
+    /**
+     * @notice Withdraws deposited collateral from the created escrow of a bid that has been successfully repaid.
+     * @param _bidId The id of the bid to withdraw collateral for.
+     */
+    function withdraw(uint256 _bidId) external {
+        BidState bidState = tellerV2.getBidState(_bidId);
+
+        require(bidState == BidState.PAID, "Loan has not been paid");
+        
+        _withdraw(_bidId, tellerV2.getLoanBorrower(_bidId));
+
+        emit CollateralClaimed(_bidId);
+    }
+
+     /**
+     * @notice Withdraws deposited collateral from the created escrow of a bid that has been successfully repaid.
+     * @param _bidId The id of the bid to withdraw collateral for.
+     * @param _recipient The address that will receive the collateral.
+     */
+    function withdrawForRecipient(uint256 _bidId, address _recipient) external {
+        BidState bidState = tellerV2.getBidState(_bidId);
+
+        require(bidState == BidState.PAID, "Loan has not been paid");
+
+        require(_msgSender() == tellerV2.getLoanBorrower(_bidId), "Not authorized");
+        
+        _withdraw(_bidId, _recipient);
+
+        emit CollateralClaimed(_bidId);
+    }
+
+    /**
+     * @notice Withdraws deposited collateral from the created escrow of a bid that has been CLOSED after being defaulted.
+     * @param _bidId The id of the bid to withdraw collateral for.
+     */
+    function lenderClaimCollateral(uint256 _bidId) external onlyTellerV2 {
+        if (isBidCollateralBacked(_bidId)) {
+            BidState bidState = tellerV2.getBidState(_bidId);
+
+            require(
+                bidState == BidState.CLOSED,
+                "Loan has not been closed"
+            );
+
+            _withdraw(_bidId, tellerV2.getLoanLender(_bidId));
+            emit CollateralClaimed(_bidId);
```

Comment:
lets remove the `CollateralClaimed` event since we are already emitting `CollateralWithdrawn` from the `_withdraw` function
```suggestion
```